src/jace/optimization.py

```diff
@@ -62,22 +74,46 @@ def jace_optimize(tsdfg: tjsdfg.TranslatedJaxprSDFG, **kwargs: Unpack[CompilerOp
 
     Args:
         tsdfg: The translated SDFG that should be optimized.
+        device: The device on which the SDFG will run on.
         simplify: Run the simplification pipeline.
-        auto_optimize: Run the auto optimization pipeline (currently does nothing)
+        auto_optimize: Run the auto optimization pipeline.
         persistent_transients: Set the allocation lifetime of (non register) transients
             in the SDFG to `AllocationLifetime.Persistent`, i.e. keep them allocated
             between different invocations.
-    """
-    # TODO(phimuell): Implement the functionality.
-    # Currently this function exists primarily for the sake of existing.
+        validate: Perform validation at the end.
+        validate_all: Perform extensive validation.
 
+    Note:
+        Currently DaCe's auto optimization pipeline is used when auto optimize is
+        enabled. However, it might change in the future. Because DaCe's auto
+        optimizer is considered unstable it must be explicitly enabled.
+    """
+    assert device in {dace.DeviceType.CPU, dace.DeviceType.GPU}
     simplify = kwargs.get("simplify", False)
     auto_optimize = kwargs.get("auto_optimize", False)
+    validate = kwargs.get("validate", DEFAULT_OPTIMIZATIONS["validate"])
```

Comment:
Why is `persistent_transients` ignored?

Reply:
This is a behavior in DaCe I do not like and with this you can disable it, at least in the future this is the plan.